pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -30,14 +30,18 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 		return err
 	}
 
-	if !strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
-		r.log.Infof("Fixing NSG from %s to %s", *subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID)
+	// if the NSG is assigned && it's the correct NSG - do nothing
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil && strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
+		return nil
+	}
 
-		subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
-		err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
-		if err != nil {
-			return err
-		}
+	// else the NSG assignment needs to be corrected
+	oldNSG := "nil"
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
+		oldNSG = *subnetObject.NetworkSecurityGroup.ID
 	}
-	return nil
+	r.log.Infof("Fixing NSG from %s to %s", oldNSG, correctNSGResourceID)
+	subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
+	err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
+	return err
```

Comment:
```suggestion
	return nil
```

Reply:
Thanks, the suggestion to return nil is not clear on its own. I was providing it in context with this suggestion: https://github.com/Azure/ARO-RP/pull/2116#discussion_r869570263